test/spec/modules/BTBidAdapte_spec.js

```diff
@@ -0,0 +1,432 @@
+            },
+            device: {
+              w: 1920,
+              h: 1280,
+              ua: 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/117.0.0.0 Safari/537.36',
+              language: 'en',
+            },
+            site: {
+              domain: 'test.com',
+              page: 'https://www.test.com/',
+              publisher: {
+                domain: 'test.com',
+              },
+            },
+          },
+          ortb2Imp: {
+            ext: {
+              tid: '12345',
+              gpid: 'path/leaderboard',
+            },
+          },
+          params: {
+            ab: true,
+            siteId: '1234567',
+            pubmatic: {
+              publisherId: '111111',
+            },
+          },
+          schain: {
+            ver: '1.0',
+            complete: 1,
+            nodes: [{ asi: 'test.com', sid: '1234567', hp: 1 }],
+          },
+          transactionId: '111111',
+          userIdAsEids: [{ source: 'test.com', uids: [{ id: '11111' }] }],
+        },
+      ];
+
+      const expectedData = JSON.stringify({
+        id: 'test12345',
+        source: {
+          tid: '111111',
+          ext: {
+            schain: {
+              ver: '1.0',
+              complete: 1,
+              nodes: [{ asi: 'test.com', sid: '1234567', hp: 1 }],
+            },
+          },
+        },
+        imp: [
+          {
+            id: 'bid12345',
+            tagid: 'leaderboard',
+            ab: true,
+            siteId: '1234567',
+            ext: { pubmatic: { publisherId: '111111' } },
+            banner: { format: [{ w: 728, h: 90 }] },
+            bidfloor: 0.03,
+            bidfloorcur: 'USD',
+            gpid: 'path/leaderboard',
+          },
+        ],
+        device: {
+          w: 1920,
+          h: 1280,
+          ua: 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/117.0.0.0 Safari/537.36',
+          language: 'en',
+        },
+        site: {
+          domain: 'test.com',
+          page: 'https://www.test.com/',
+          publisher: { domain: 'test.com' },
+        },
+        user: {
+          ext: {
+            consent: 'gdprString',
+            eids: [{ source: 'test.com', uids: [{ id: '11111' }] }],
+          },
+        },
+        regs: { ext: { us_privacy: 'uspString' } },
+        at: 1,
+        ext: { tid: '12345', gpid: 'path/leaderboard' },
+      });
+
+      const requests = spec.buildRequests(validBidRequests, bidderRequest);
+
+      expect(requests).to.be.an('array');
+      expect(requests).to.have.lengthOf(validBidRequests.length);
+      expect(requests[0].method).to.equal('POST');
+      expect(requests[0].url).to.equal(ENDPOINT_URL);
+      expect(requests[0].data).to.equal(expectedData);
+      expect(requests[0].bids).to.equal(validBidRequests);
+    });
+
+    it('should build post request ortb2 fields are not present', () => {
```

Comment:
Should this be something like: "should build post request **even when** ortb2 fields are not present"?